iXBRLViewerPlugin/iXBRLViewer.py

```diff
@@ -154,26 +154,28 @@ def addConcept(self, concept):
 
     def treeWalk(self, rels, item, indent = 0):
         for r in rels.fromModelObject(item):
-            self.treeWalk(rels, r.toModelObject, indent + 1)
+            if r.toModelObject is not None:
+                self.treeWalk(rels, r.toModelObject, indent + 1)
 
     def getRelationships(self):
         rels = {}
 
         for baseSetKey, baseSetModelLinks  in self.dts.baseSets.items():
             arcrole, ELR, linkqname, arcqname = baseSetKey
-            if arcrole == XbrlConst.summationItem and ELR is not None:
+            if (arcrole == XbrlConst.parentChild or arcrole == XbrlConst.summationItem) and ELR is not None:
                 self.addELR(ELR)
                 rr = dict()
                 relSet = self.dts.relationshipSet(arcrole, ELR)
                 for r in relSet.modelRelationships:
-                    fromKey = self.nsmap.qname(r.fromModelObject.qname)
-                    rel = {
-                        "t": self.nsmap.qname(r.toModelObject.qname),
-                    }
-                    if r.weight is not None:
-                        rel['w'] = r.weight
-                    rr.setdefault(fromKey, []).append(rel)
-                    self.addConcept(r.toModelObject)
+                    if r.fromModelObject is not None and r.toModelObject is not None:
```

Comment:
I'm not sure how it is possible to get a relationship which doesn't have a from or to object in Arelle, but if it is, then this guard is a useful addition.